### ISSUE TITLE
Console Exports: Due to the chosen wording, people mistakenly think out of the box support for consoles is impossible

### ISSUE THREAD

**Issue by golddotasksquestions** [godotengine/godot-docs#5625]
**Your Godot version:** 3.4.2.stable

**Issue description:**
I've had this conversation now ... I don't know how many times. It's one of these things that come up like clockwork in the community channels:

A lot of people mistakenly seem to think Godot will never have or cannot have out of the box support for consoles like Unity or Unreal does due to Godot being open source.

This is of course completely wrong and merely a matter of incentive and ironing out relevant details.

It seems to me a lot of these people get the idea from this part of the docs page which I am often linked to in these discussions: 

> The reason other consoles are not officially supported are:
> 
>     To develop for consoles, one must be licensed as a company. As an open source project, Godot does not have such a legal figure.
>     Console SDKs are secret and covered by non-disclosure agreements. Even if we could get access to them, we could not publish the platform-specific code under an open source license.
>     Consoles require specialized hardware to develop for, so regular individuals can't create games for them anyway.

While they are not factually wrong, these point makes it apparently sound to many as if there is no legal possible way for Godot users to have the same convenience and cost saving of premade export templates like Unity or Unreal users have.
The text tries to explain this, but phrases like 

>we could not publish the platform-specific code under an open source license.

make is sound as if releasing platform-specific (=proprietary) code under open source license would be a necessity in order to provide Godot users the same console support as Unity or Unreal users. [This is obviously not the case.](https://www.reddit.com/r/godot/comments/svp9ll/comment/hxp3ueu/?utm_source=share&utm_medium=web2x&context=3). 

At no point Godot would ever have to open source proprietary code in order to get the same console support as Unity or Unreal. Of course there are a lot of other questions that need to be answered and probably also legal details to be discussed if console manufactures in fact have the interest to partner with Godot in some shape or form. Such as who pays for the development and maintenance of the export templates and where are they hosted. But it's far from impossible like the doc page seem to suggest for many readers.

**URL to the documentation page (if already existing):**
https://docs.godotengine.org/en/stable/tutorials/platform/consoles.html#official-support

**Comment by pycbouh** [comment]
> Due to the chosen wording, people mistakenly think out of the box support for consoles is impossible

This is correct, out of the box support for Godot is impossible. This is mentioned in the thread that you've linked: you need to go through a third party, or get into direct contact with a console vendor and port the platform code for their platform yourself. But it is impossible to ship Godot with templates for consoles. Until vendors come forth and permit it, that is. And that is what the open source license quote is about.

**Comment by golddotasksquestions** [comment]
@pycbouh 
What you read, know and understand as "currently impossible to directly ship Godot with console export templates", others wrongly interpret as "it is forever impossible to have the same convenience as other engines due to the Godot engine being open source".

This is a fundamental misunderstand and a false tale which I regular see popping up on various community channels and threads.

The source of this tale is for a large part the wording on this docs page.

The consequence of this misunderstanding and false tale people spread is that they think "Godot will never be able to support consoles like Unreal or Unity out of the box". While it very well can. Godot will never "ship" with console export templates, or make those opensource, but this in no way means we can't have a partnership with console manufactures and arrange a deal to allow the same out of the box convenience, therefore "officially supported by console manufacture XYZ".

The current wording of this page leads to more people who could potentially archive greatness with Godot to decide against it. But we need devs who aim for greatness! It's great to support hobbyist and beginners, but it's the greatness of successful popular projects that bring in attention and with it more support and sponsorship. Support and sponsorships beginners and hobbyists profit just the same.

**Comment by pycbouh** [comment]
> This is a fundamental misunderstanding and a false tale

Telling people that the engine at some point _might_ offer first-party export templates is misleading and false marketing. There is no false tale. The docs state facts, and people are correctly assuming that as it stands, Godot cannot offer support for console exports out of the box, for the reasons outlined in the docs.

I'm thankful that you want Godot to be presented in a better light and worry about the image of the engine, but in this case we state facts as they are, and there is no misunderstanding from users who you are trying to correct.

**Comment by golddotasksquestions** [comment]
>  and there is no misunderstanding from users who you are trying to correct.

There is. Thinking "**it will never happen because it is legally not possible**" is the misunderstanding.

**Comment by pycbouh** [comment]
> This converstation with you makes it even more apparent. If even long term contributors can't tell the difference, no surprise someone who is much less involved will understand it. Especially since the docs are worded as they are.

What difference are you speaking about? Docs state in no uncertain terms that it is not possible to provide official Godot support for console exports and lists several reasons for it. One of which is licensing issues, which are and will always be an issue.

If vendors ever decide to provide first-party export templates, then vendor will provide them, not Godot, not as a part of Godot's distribution. To a degree, this is what happens with other engines too. Vendors provide ways to export for their platforms, provided the developer can confirm they are licensed to export for the said platform. It's not the engine developers who provide those export templates.

If something changes in the future, those statements would still be correct. But we cannot promise something that is not true and may never be true, because there is little to no precedent for something like this happening.

**Comment by Calinou** [comment]
To be precise, there *are* open source game frameworks out there like MonoGame that provide console support to licensed developers (and publicly advertise this). There's also open source software that publicly advertises console support, such as RenderDoc on Nintendo Switch. Again, the ports are provided to licensed developers *only*.

Maybe the introduction should be reworded to remove the ambiguity. It's not because Godot is open source that it doesn't have console ports, it's because there is no legal structure to provide such console ports to licensed developers (and ensure support and maintenance for those console ports).

**Comment by pycbouh** [comment]
> To be precise, there _are_ open source game frameworks out there like [MonoGame](https://www.monogame.net/) that provide console support to licensed developers (and publicly advertise this). There's also open source software that publicly advertises console support, such as [RenderDoc](https://renderdoc.org/) on Nintendo Switch. Again, the ports are provided to licensed developers _only_.

And, according to the MonoGame docs, you get access through the vendor:

> Platforms for systems that require registration such as PlayStation and Xbox are not publicly available. To get access to those platforms, please contact your console account manager(s).

**Comment by golddotasksquestions** [comment]
> You are in fact talking about changing wording related to something that has not happened and has no indication of happening. If you believe I'm mistaken, can you provide a suggestion, which statement would you change and how?

I would straight up remove the whole "Official Support" paragraph. 1000$ is also pretty worthless information. It's highly dependent on the game and console you want to port and lot's of other factors. So I would also remove that.

[The first paragraph](https://docs.godotengine.org/en/stable/tutorials/platform/consoles.html#console-publishing-process) already explained the process very well, so does the Note and the paragraph below [about third party support](https://docs.godotengine.org/en/stable/tutorials/platform/consoles.html#third-party-support).
The only thing here missing here I think is a mention of export templates and their role. I would also like to better explain the porting process and why it is a good idea to outsource.

Here is the PR:
https://github.com/godotengine/godot-docs/pull/5627
